include/LeMonADE/feature/FeatureWall.h

```diff
@@ -0,0 +1,310 @@
+    
+    const VectorInt3 getNormal() const {
+        return normal;
+    }
+    
+    VectorInt3 setNormal(uint32_t norX_, uint32_t norY_, uint32_t norZ_) {
+        normal.setAllCoordinates(norX_,norY_,norZ_);
+    }
+    
+private:
+    
+    VectorInt3 base;
+    
+    VectorInt3 normal;
+    
+};
+
+
+
+class FeatureWall: public Feature
+{
+public:
+    
+    //constructor
+    FeatureWall() {}
+    
+    //destructor
+    virtual ~FeatureWall(){}
+    
+    //read and write wall in/from bfm-file
+    template<class IngredientsType> 
+    void exportRead(FileImport<IngredientsType>& fileReader);
+    
+    template<class IngredientsType> 
+    void exportWrite(AnalyzerWriteBfmFile<IngredientsType>& fileWriter) const;
+    
+    //checks if move is allowed by the Metropolis-criterion.
+	template<class IngredientsType> 
+	bool checkMove(const IngredientsType& ingredients,MoveLocalSc& move);
+    
+    template<class IngredientsType>
+    bool checkMove(const IngredientsType& ingredients,MoveAddScMonomer& addmove); 
+    
+    template<class IngredientsType>
+    void synchronize(const IngredientsType& ingredients);
+    
+    std::vector<Wall> getWalls() const
+    {
+        return walls;
+    }
+    
+    void addWall(Wall wall)
+    {
+        walls.push_back(wall);
+    }
+    
+    void clearAllWalls()
+    {
+        walls.clear();
+    }
+    
+    
+private:
+    
+    std::vector<Wall> walls;
+        
+};
+
+
+
+template <class IngredientsType>
+class WriteWall: public AbstractWrite<IngredientsType>
+{
+    public:
+        
+        WriteWall(const IngredientsType& src):AbstractWrite<IngredientsType>(src){this->setHeaderOnly(true);}
+  
+        void writeStream(std::ostream& strm)
+        {
+            const IngredientsType& ingredients=(this->getSource());
+            
+            for (size_t i=0; i<ingredients.getWalls().size(); i++) {
+                strm << "#!wall: " << ingredients.getWalls()[i].getBase().getX() << " " << ingredients.getWalls()[i].getBase().getY() << " " << ingredients.getWalls()[i].getBase().getZ() << "\t" << ingredients.getWalls()[i].getNormal().getX() << " " << ingredients.getWalls()[i].getNormal().getY() << " " << ingredients.getWalls()[i].getNormal().getZ() << "\n";
+            }
+            strm << "\n\n";
+        }
+};
+
+
+
+template < class IngredientsType >
+class ReadWall : public ReadToDestination < IngredientsType >
+{
+    public:
+        
+        ReadWall(IngredientsType& destination):ReadToDestination< IngredientsType > (destination){}
+
+        void execute();
+};
+
+
+
+
+
+
+
+/* checkMove */
+
+template<class IngredientsType>
+bool FeatureWall::checkMove(const IngredientsType& ingredients, MoveLocalSc& move) 
+{	
+    uint32_t counter(0);
+    
+	for (size_t i = 0; i < ingredients.getWalls().size(); i++) { //check all walls in the system
+        
+		uint8_t direction;
+        VectorInt3 unitX(1,0,0);
+        VectorInt3 unitY(0,1,0);
+        VectorInt3 unitZ(0,0,1);
+		
```

Comment:
I think the best way to do this, is in the setter functions and on any places where one can modify the normal vector. So one can be shure to have a correct normal vector from the beginning. I will add this and also some doxygen like comments.